### PR TITLE
Week 3 ec

### PR DESCRIPTION
**Pull request by EddieCarrion**
# Submitting Week 3 Assessment

### ✅ Completed Files

Mark the following as confirmation that the files in this repository are complete.

- [x ] Coding Challenges
- [ x] Interview Questions
- [ x] React Debugging

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here: I am having a tough time retaining the information and spending a lot of time on each code challenge. Any tips to improve on retaining the information?

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -14,19 +14,87 @@
...
+
 // b) Create the function that makes the test pass.
 
+// Process
+// First I will create a function named "fib" that takes in a number as a parameter
+// The I will create a conditional statement to check if the number passed in the parameter are less than or equal to 2
+// Next I will create an array that contains the first two elements of the sequence
+// Next I will create a for loop that begins at the 2 index that does not go over the number given as the parameter
+// Next I will use the push method to add the previous two numbers in the array.
+// Finally I will call on the variable to make the test pass
```
Comment (SunkissedQueen) [inline code-challenges.test.js:51]: Great job getting a function to pass the infamous fibonacci sequence. Now that you have working logic, make sure that your pseudocode conveys what is occurring in your function. 
This is a suggestion ---> Ensure you are speaking more about the data than finding a tool that can help you. If you know of a tool, then mention it after talking about your data. For instance:
Method focused: `Then I will create a conditional statement to check if the number passed in the parameter are less than or equal to 2`
Data focused: `if the number is less or equal to 2, then return a string that states "number must be great than 2" Tool to be used: conditional statements`

File: code-challenges.test.js [hunk 2]
```diff
@@ -14,19 +14,87 @@
...
+  }else {
+    let fibSequence = [1,1]
+    for (let i=2; i<number; i++) {
+      fibSequence.push(fibSequence[i-1]+fibSequence[i-2])
+    } 
+    return fibSequence
+  }
+}
+
+// Test passed
```
Comment (SunkissedQueen) [inline code-challenges.test.js:65]: great job remembering all the edge cases

File: code-challenges.test.js [hunk 3]
```diff
@@ -14,19 +14,87 @@
...
+// I will create two expects statments and use the provided objects as arguments and return the expected sequence
+
+describe ("studyMinutes", () => {
+  it ("will take in an object and return an array of the number of minutes studied from least to greatest", () => {
+      expect (studyMinutes(studyMinutesWeek1)).toEqual([15, 15, 20, 30, 30, 60, 90])
+      expect (studyMinutes(studyMinutesWeek2)).toEqual([10, 15, 20, 45, 60, 65, 100])
+      
+      
+  })
+})
```
Comment (SunkissedQueen) [inline code-challenges.test.js:89]: nice structure and indentation

File: code-challenges.test.js [hunk 4]
```diff
@@ -51,10 +119,51 @@ const studyMinutesWeek2 = {
...
+// Next take the values from the object and store them in an array named "minutes"
+// Next sort the values from least to greatest and store them into a variable named "organizedMinutes"
+// Finally return the assorted array that contain the organized minutes
+
+
+const studyMinutes = (object) =>{
+  let minutes = Object.values(object)
+  let organizedMinutes = minutes.sort((a,b)=> a - b)
+  return organizedMinutes
+}
```
Comment (SunkissedQueen) [inline code-challenges.test.js:133]: nice work. A possible refactor could be a return statement that is chaining the methods

File: code-challenges.test.js [hunk 5]
```diff
@@ -51,10 +119,51 @@ const studyMinutesWeek2 = {
...
+// Snapshots:   0 total
+// Time:        0.14 s, estimated 1 s
+
 // --------------------3) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
 
 // a) Create a test with expect statements for each of the variables provided.
 
+// I will create a "describe" statement that contains the function named "sum".
+// I will create an "it" statement that will explain how the function will take in an array and return the accumulated sum.
+// I will create three expects statments and use the provided variables as arguments and return the expected accumulated sum.
```
Comment (SunkissedQueen) [inline code-challenges.test.js:149]: love that you provide pseudocode for the test

File: code-challenges.test.js [hunk 6]
```diff
@@ -65,3 +174,29 @@ const accountTransactions3 = []
 // Expected output: []
 
 // b) Create the function that makes the test pass.
+
+// Process
+// First I will create a function named "sum" that takes in an array as a parameter
+// Next create variable that contains an empty array that will store the accumulated sum
+// Next create another variable that will hold the accumulated sum after each step
+// Next create a for loop that goes the entire length of the array that adds each element to the "total" variable and adds that value to the array
```
Comment (SunkissedQueen) [inline code-challenges.test.js:182]: need a bit more explanation about what the for loop will provide to you and what actions you will take with that data

File: code-challenges.test.js [hunk 7]
```diff
@@ -65,3 +174,29 @@ const accountTransactions3 = []
...
+  
+    let results = []
+    let total = 0
+    for (let i=0; i<arr.length; i++) {
+      total += arr[i]
+      results.push(total)
+    } 
+    return results
+  
+}
```
Comment (SunkissedQueen) [inline code-challenges.test.js:195]: woohoo

File: interview-questions.md [hunk 8]
```diff
@@ -8,48 +8,60 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is JSX? What is one syntax difference between HTML and JSX?
 
-Your answer:
+Your answer: Jsx is a tool helps display javascript logic on the web page.
 
-Researched answer:
+Researched answer: Jsx is an extention to javascript that acts alot like HTML. Jsx is the foundation of React components that allows it to make user interfaces.
+
```
Comment (SunkissedQueen) [inline interview-questions.md:14]: remember the interview is about sharing as much knowledge as possible. More content can be sharing your experience when you first used JSX in React

File: interview-questions.md [hunk 9]
```diff
@@ -8,48 +8,60 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+Resource - Syllabus
 
 1. What is yarn? What file(s) are modified when you run the command yarn in your terminal?
 
-Your answer:
+Your answer: Yarn organizes and stores an abundance of code snippets.
+
+Researched answer: Yarn is a Package that is responsible for storing node modules in Javascript
```
Comment (SunkissedQueen) [inline interview-questions.md:21]: Eddie, you are a great storyteller. Please use that skill set in the interview section

File: interview-questions.md [hunk 10]
```diff
@@ -8,48 +8,60 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
-Researched answer:
+Resource - Syllabus
 
 3. What is a React component?
 
-Your answer:
+Your answer: React components are pieces of Javascript coding that get compartmentalized 
+
+Researched answer: React components are specific functions that return javascript logic to the browser.
```
Comment (SunkissedQueen) [inline interview-questions.md:29]: yes, I am asking for more content

File: interview-questions.md [hunk 11]
```diff
@@ -8,48 +8,60 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+
+Researched answer: React components are specific functions that return javascript logic to the browser. 
 
-Researched answer:
+Resource - Syllabus
 
 4. What is the difference between state values and props in React?
 
-Your answer:
+Your answer: State values are piece of data that can change depending on how th user is interacting. Props are pieces of data that do not change.
```
Comment (SunkissedQueen) [inline interview-questions.md:35]: I like where this was headed more so than the researched answer.More content can be to share examples of how a state value look and how it looks to pass it as a prop. You can even mention what is needed to perform the actions.

File: interview-questions.md [hunk 12]
```diff
@@ -8,48 +8,60 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+Researched answer: State values are variables that dictate how a component will behave. Props have the capacity pass data from one components to another. 
+
+Resource - Syllabus
 
 5. What is the DOM?
 
-Your answer:
+Your answer: DOM is the dislay you see on the browser. It is the end result of your code.
+
+Researched answer: DOM stands for Document Object Model, it is the visual reprensetation of what was done by the web developers.
```
Comment (SunkissedQueen) [inline interview-questions.md:45]: Too much space is left on the table. Let's feed the interviewer more information about your experience with the tool. You can share your experience and share examples of how the code looks as html and talk about how that is presented on the screen.

File: interview-questions.md [hunk 13]
```diff
@@ -8,48 +8,60 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
-Researched answer:
+Resource - Syllabus
 
 6. STRETCH: Which is the difference between a div tag and a span tag?
 
-Your answer:
+Your answer:The div tag allows you to contain and divide your information. I am not sure what the span tag does, I am not familiar with it yet.
+
+Researched answer: The div tag allows you to create divisions of information on the webpage. The span tag works as an inline container that is used for styling purposes.
```
Comment (SunkissedQueen) [inline interview-questions.md:53]: Appreciate that you answered the stretch.

File: interview-questions.md [hunk 14]
```diff
@@ -8,48 +8,60 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
-Researched answer:
+Resource - geeksforgeeks.org
 
 ## Looking Ahead: Terms for Next Week
```
Comment (SunkissedQueen) [inline interview-questions.md:58]: hopefully this helps with understanding lectures

File: react-debugging.html [hunk 15]
```diff
@@ -65,11 +65,17 @@
         // No need to change anything ABOVE this line ^
 
         return (
+
+        // The elements in this section were not wrapped properly, a Div container was missing
+        // The onCLick event had no function to call on (line 73)
```
Comment (SunkissedQueen) [inline react-debugging.html:70]: great job tacking the debugging

**Review by SunkissedQueen — APPROVED** [review]
Eddie, you are approved to push these changes. Appreciate the efforts you are taking. I can see that you have been studying. You are certainly performing as an Advanced Beginner. Please review the notes and suggestions. Next assessment on the pseudocode please be mindful to talk about what you are doing to the data and include more content on the interview questions.
